mountpoint-s3/src/fuse/session.rs

```diff
@@ -52,10 +52,22 @@ impl FuseSession {
                 .name("fuse-worker-waiter".to_owned())
                 .spawn(move || {
                     for thd in workers {
-                        let result = thd.join().unwrap();
-                        if let Err(e) = result {
-                            error!("fuse worker thread failed: {e:?}");
-                        }
+                        let thread_name = thd
+                            .thread()
+                            .name()
+                            .expect("worker threads should have names")
+                            .to_owned();
+
+                        match thd.join() {
+                            Err(_panic_param) => {
```

Comment:
You could use `downcast_ref::<String>()` on the parameter to get the message to report.

Reply:
Makes sense. I found that it won't work if the panic was called with a string literal only (i.e. it was called with `&str`), so its a little more complex but I think OK.